### PR TITLE
After revolutionaries win, one living headrev (if any are able to) will be allowed to rename the station with a revolutionary banner

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Really everything is in the title here. Headrevs win, if any are not in crit, one will be given a banner. They have 5 minutes to rename the station just like the initial charter.

## Why It's Good For The Game

We need more tangible differences between a revolutionary station and a non revolutionary one. Giving it a rev'd up name seems good

## Changelog
:cl:
add: One headrev is tasked with renaming the station when revolutions are successful
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
Oh yeah and obviously getting a revolutionary flag is a pretty dope souvenir

File: code/game/objects/items/charter.dm [hunk 1]
```diff
@@ -121,3 +121,37 @@
...
+	addtimer(VARSET_CALLBACK(src, timesup, TRUE), 5 MINUTES)
+
+/obj/item/station_charter/revolution/attack_self(mob/living/user)
+	if(timesup && !used)
+		to_chat(user, "<span class='warning'>You have lost the victorious fervor to declare a new name.</span>")
+		return
+	. = ..()
+
+/obj/item/station_charter/revolution/rename_station(designation, uname, ureal_name, ukey)
+	. = ..()
```
Comment (tralezab) [inline code/game/objects/items/charter.dm:149]: cut this 💅

Comment (Mothblocks) [reply code/game/objects/items/charter.dm:149]: is this not the place admins get allowed to veto names and such?

Comment (tralezab) [reply code/game/objects/items/charter.dm:149]: This is after a name is accepted, it's calling parent which is also setting the name and announcing it. Eek!

**Comment by nianjiilical** [comment]
would it be possible to automatically send a message to other stations just to rub it in

**Comment by Farquaar** [comment]
> 
> 
> would it be possible to automatically send a message to other stations just to rub it in

Please continue working as normal.. There is nothing wrong with your sister station in the Bagil sector. Do not contact them. Their communications array is undergoing repairs. Do not read any absurd or erroneous messages you may receive from them.

File: code/modules/antagonists/revolution/revolution.dm [hunk 2]
```diff
@@ -383,13 +383,28 @@
...
+				if(!charter_given && rev_mind.current && rev_mind.current.stat == CONSCIOUS)
+					charter_given = TRUE
+					var/obj/structure/closet/supplypod/bluespacepod/syndicate/pod = new()
+					pod.explosionSize = list(0,0,0,0)
+					new /obj/item/station_charter/revolution(pod)
+					var/turf/landing_location = get_turf(rev_mind.current)
+					new /obj/effect/pod_landingzone(landing_location, pod)
+					to_chat(rev_mind.current, "<span class='hear'>You hear something crackle in your ears for a moment before a voice speaks. \
+					\"Please stand by for a message from your benefactor. Message as follows, provocateur. \
+					<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\"</span>")
```
Comment (Mothblocks) [inline code/modules/antagonists/revolution/revolution.dm:407]:
```suggestion
						\"Please stand by for a message from your benefactor. Message as follows, provocateur. \
						<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\"</span>")
```

File: code/modules/antagonists/revolution/revolution.dm [hunk 3]
```diff
@@ -383,13 +383,28 @@
...
+				LAZYADD(rev_mind.special_statuses, "<span class='bad'>Former revolutionary</span>")
+			else
+				LAZYADD(rev_mind.special_statuses, "<span class='bad'>Former head revolutionary</span>")
+				if(!charter_given && rev_mind.current && rev_mind.current.stat == CONSCIOUS)
+					charter_given = TRUE
+					var/obj/structure/closet/supplypod/bluespacepod/syndicate/pod = new()
+					pod.explosionSize = list(0,0,0,0)
+					new /obj/item/station_charter/revolution(pod)
+					var/turf/landing_location = get_turf(rev_mind.current)
+					new /obj/effect/pod_landingzone(landing_location, pod)
```
Comment (Mothblocks) [inline code/modules/antagonists/revolution/revolution.dm:404]: We really need a separate proc for this, feels like we're doing this a lot. I don't know if it's a requirement here, but would be nice to have.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
One pedantic change, and another that I'd like you to think on to see if you want to do it in this PR.

**Pull request merged**